File: Frontend/src/lib/services/metrics.ts
import { invoke } from "@tauri-apps/api/core";
import { Metric } from "@/interfaces/entities/metric";
import { Result } from "@/types/result";

// db
async function select_metrics(env: string): Promise<Result<Metric[]>> {
    try {
        const data = await invoke<Metric[]>("select_metrics", {
            env,
        });

        return { ok: true, data };
    } catch (error) {
        return { ok: false, error };
    }
}

// runtime
async function get_active_metric(
    env: string,
): Promise<Result<Metric | null>> {
    try {
        const data = await invoke<Metric | null>("get_active_metric", {
            env,
        });

        return { ok: true, data };
    } catch (error) {
        return { ok: false, error };
    }
}

export const metricService = {
    // db
    selectAll: (env: string): Promise<Result<Metric[]>> => select_metrics(env),

    // runtime
    getActive: (env: string): Promise<Result<Metric | null>> =>
        get_active_metric(env),
};

File: Frontend/src-tauri/src/command/metrics.rs
use crate::{
    core::{get_active_metric_core, select_metrics_core},
    models::entities::critical_metrics::Model,
};

#[tauri::command]
pub async fn select_metrics(env: String) -> Result<Vec<Model>, String> {
    select_metrics_core(env).await
}

#[tauri::command]
pub async fn get_active_metric(env: String) -> Result<Option<Model>, String> {
    get_active_metric_core(env).await
}
